src/oss-find-squats/FindSquatsTool.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        HttpClient client;
+        static async Task Main(string[] args)
+        {
+            var findSquatsTool = new FindSquatsTool();
+            await findSquatsTool.ParseOptions<Options>(args).WithParsedAsync(findSquatsTool.RunAsync);
+        }
+
+        public async Task RunAsync(Options opts)
+        {
+            var results = new List<(IEnumerable<string> rule, PackageURL purl)>();
+
+
+            SarifLog sarifLog = new SarifLog();
+            sarifLog.Version = SarifVersion.Current;
+            Run runItem = new Run();
+            runItem.Tool = new Tool();
+
+            runItem.Tool.Driver = new ToolComponent();
+            if (Assembly.GetEntryAssembly() is Assembly entryAssembly)
+            {
+                runItem.Tool.Driver.Name = entryAssembly.GetName().Name;
+
+                runItem.Tool.Driver.FullName = entryAssembly.GetCustomAttribute<AssemblyProductAttribute>()?
+                                                     .Product;
+
+                runItem.Tool.Driver.Version = entryAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?
+                                                    .InformationalVersion;
+            }
+
+            var descriptors = new List<ReportingDescriptor>();
+
+            foreach (var mutation in gen.Mutations)
+            {
+                ReportingDescriptor sarifRule = new ReportingDescriptor();
+                sarifRule.Name = mutation.Method.Name;
+                sarifRule.DefaultConfiguration = new ReportingConfiguration() { Enabled = true };
+            }
+
+            runItem.Tool.Driver.Rules = descriptors;
+
+            var sarifResults = new List<Result>();
+
+            foreach (var target in opts.Targets ?? Array.Empty<string>())
+            {
+                var purl = new PackageURL(target);
+                if (purl.Name is not null && purl.Type is not null)
+                {
+                    var manager = ProjectManagerFactory.CreateProjectManager(purl, null);
+                    if (manager is not null)
+                    {
+                        var mutationsDict = gen.Mutate(purl.Name);
+
+                        foreach ((var candidate, var rules) in mutationsDict)
+                        {
+                            Logger.Info($"Checking {candidate}.");
+                            var candidatePurl = new PackageURL(purl.Type, candidate);
+                            var versions = await manager.EnumerateVersions(candidatePurl);
+
+                            if (versions.Any())
+                            {
+                                Logger.Warn($"{candidate} package exists. Potential squat.");
+                                Result res = new Result();
+                                res.Locations = new List<Location>();
+                                Location loc = new Location();
+
+                                loc.LogicalLocation = new LogicalLocation()
```

Comment:
Technically this is a physical location. Checkout the BuildPurlLocation() function in SarifOutputBuilder class for this.